### PR TITLE
[fix bug2689]cannot maintain disconnected primary storage

### PR DESCRIPTION
**Pull request by HeathHose**

### CODE REVIEW

**Comment by HeathHose** [comment]
@ZStack-Robot 
cannot maintain disconnected primary storage
https://github.com/zstackio/issues/issues/2689

File: test/src/test/groovy/org/zstack/test/integration/storage/primary/nfs/NfsDisconnectedCase.groovy [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    void testMaintainDisconnectedPrimaryStorage(){
+        PrimaryStorageInventory ps = env.inventoryByName("nfs")
+        KVMAgentCommands.ConnectCmd cmd = null
+        env.simulator(KVMConstant.KVM_CONNECT_PATH){HttpEntity<String> entity,EnvSpec spec ->
+            def rsp = new KVMAgentCommands.ConnectResponse()
+            rsp.success = false
+            return rsp
+        }
+        try{
+            reconnectPrimaryStorage {
```
Comment (zxwing) [inline test/src/test/groovy/org/zstack/test/integration/storage/primary/nfs/NfsDisconnectedCase.groovy]: gai

File: test/src/test/groovy/org/zstack/test/integration/storage/primary/nfs/NfsDisconnectedCase.groovy [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                uuid = ps.uuid
+            }
+        }
+        expect(ApiMessageInterceptionException.class){
+            ChangeHostStateAction action = new ChangeHostStateAction()
+            action.uuid = ps.uuid
+            action.stateEvent = PrimaryStorageStateEvent.maintain.toString()
+            action.sessionId = loginAsAdmin()
+            action.call()
```
Comment (zsyzsyhao) [inline test/src/test/groovy/org/zstack/test/integration/storage/primary/nfs/NfsDisconnectedCase.groovy]: here need to fix the action invoke.

**Comment by zxwing** [comment]
no such logic

**Comment by HeathHose** [comment]
## Analysis
The action of maintain primary storage is to stop vms and if primary storage is disconnected the vm have been stopped. so it's ok that maintian disconnnect primary storage
On the other hander, the action of maintain host is to migrate vm but zstack cannot migrate vm on the disconnected host.so it isn't allowed that maintian disconnect host

**Pull request closed without merging**